src/writer.rs

```diff
@@ -0,0 +1,746 @@
+// Copyright 2021 The Chromium OS Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+//! This module writes Flattened Devicetree blobs as defined here:
+//! <https://devicetree-specification.readthedocs.io/en/stable/flattened-format.html>
+
+use std::collections::BTreeMap;
+use std::convert::TryInto;
+use std::ffi::CString;
+use std::mem::size_of;
+
+use thiserror::Error as ThisError;
+
+use crate::{FDT_BEGIN_NODE, FDT_END, FDT_END_NODE, FDT_MAGIC, FDT_PROP};
+
+#[derive(ThisError, Debug)]
+pub enum Error {
+    #[error("Properties may not be added after a node has been ended")]
```

Comment:
Since we don't expect this code base to significantly grow, what do you say about removing thiserror as a dependency? I think we've been a bit too permissive with non essential dependency in library crates so far, but I am interesting to hear out your thoughts on the subject as well.

Reply:
Sounds reasonable - the crosvm code base has started migrating to it for new code, which is why it ended up here.  I have replaced it with a fmt::Display implementation instead.